src/openrct2/drawing/TTFSDLPort.cpp

```diff
@@ -388,7 +388,7 @@ static TTF_Font* TTF_OpenFontIndexRW(FILE* src, int freesrc, int ptsize, long in
         return NULL;
     }
 
-    font = static_cast<TTF_Font*>(malloc(sizeof *font));
+    font = new TTF_Font;
```

Comment:
I was actually going to, but realized that `font` is actually returned from this function, so that would require changing the function interface and tracking the ownership (can it really be unique?), so it would be more intrusive change.
And I figured that `new` is still better than `malloc()`

Reply:
You are not wrong about the scope of this PR however it looks like this is also a memory leak, should have become obvious to me given you haven't refactored free to delete elsewhere since it returns the pointer. Please revert the new/delete changes as there is no benefit to it and it should be refactored to return unique_ptr which makes ownership clear then. If you like you can do a follow up PR cleaning up that mess.

Reply:
tbh `free(font);` has been replaced with `delete font;` below too, or is it just one of the possible places where it gets deleted and the rest has not been replaced?

Reply:
That pointer escapes the function by returning it

Reply:
> Edit: Nevermind you are right it eventually calls TTF_CloseFont, definitely should use unique_ptr instead.

I am really not sure about that.

As I said before, to convert from raw pointer to the `unique_ptr` you need to have well defined ownership of the object, so I had a look at the the current code to try to understand it a bit.

From my quick inspection, it looks like allocated `font`s are kept in the global variable `gCurrentTTFFontSet` (which is actually an array of structs holding the allocated font). Since global variables never go out of scope, the destructor of the smart pointer would never be called to release the current object.
The mentioned `TTF_CloseFont` is ultimately (or indirectly) called by the `TTFDispose()` function, which in turn is called by some `Load.*Font()` functions, which seem to manually release some resources before setting them up again.

So, unless I horribly misunderstood something, it looks like the current implementation of font management is meant to rely on manual memory management, and it cannot be easily changed to adopt the smart pointers, without some serious redesign of the whole thing.

With the above in mind, my proposal is to either:
- abandon this part of the PR and leave the existing malloc/free pairs until the fonts code is refactored so `unique_ptr`s can be used, or
- accept my current changes to `new` and `delete` which do not resolve anything ownership-wise, but are still some improvement over `malloc()` and `free()` IMHO.

Please feel free to correct me if I got anything wrong (which I likely have).

Reply:
If it was up to me, I'd keep `new` and `delete` and then sort out the lifetime management in a follow-up PR.

Reply:
The point of getting rid of malloc/free is to make ownership clear, new/delete has the same deficit.

Reply:
OK, so I reverted those changes back to the original malloc and free, although I would like to point out, that IMHO just mere casting avoidance would be good enough reason for switching to new/delete.